integration-tests/smoke/ccip_test.go

```diff
@@ -276,3 +277,460 @@ func TestTokenTransfer(t *testing.T) {
+		MCMSConfig:     ccdeploy.NewTestMCMSConfig(t, e),
+		OCRSecrets:     deployment.XXXGenerateTestOCRSecrets(),
+	})
+	require.NoError(t, err)
+	require.NoError(t, e.ExistingAddresses.Merge(output.AddressBook))
+	// Get new state after migration and mock USDC token deployment.
+	state, err = ccdeploy.LoadOnchainState(e)
+	require.NoError(t, err)
+
+	// Deploy the token to test transferring
+	srcToken, _, dstToken, _, err := ccdeploy.DeployTransferableToken(
+		lggr,
+		tenv.Env.Chains,
+		tenv.HomeChainSel,
+		tenv.FeedChainSel,
+		state,
+		e.ExistingAddresses,
+		"MY_TOKEN",
+	)
+	require.NoError(t, err)
+
+	// Ensure capreg logs are up to date.
+	ccdeploy.ReplayLogs(t, e.Offchain, tenv.ReplayBlocks)
+
+	// Apply the jobs.
+	for nodeID, jobs := range output.JobSpecs {
+		for _, job := range jobs {
+			// Note these auto-accept
+			_, err := e.Offchain.ProposeJob(ctx,
+				&jobv1.ProposeJobRequest{
+					NodeId: nodeID,
+					Spec:   job,
+				})
+			require.NoError(t, err)
+		}
+	}
+
+	// Add all lanes
+	require.NoError(t, ccdeploy.AddLanesForAll(e, state))
+	// Need to keep track of the block number for each chain so that event subscription can be done from that block.
+	startBlocks := make(map[uint64]*uint64)
+	// Send a message from each chain to every other chain.
+	expectedSeqNum := make(map[uint64]uint64)
+
+	// Mint 2 tokens to be transferred
+	twoCoins := new(big.Int).Mul(big.NewInt(1e18), big.NewInt(2))
+	tx, err := srcToken.Mint(
+		e.Chains[tenv.HomeChainSel].DeployerKey,
+		e.Chains[tenv.HomeChainSel].DeployerKey.From,
+		new(big.Int).Mul(twoCoins, big.NewInt(10)),
+	)
+
+	if err != nil {
+		fmt.Printf("Error minting token: %v\n", err)
+	}
+
+	// Confirm that mint tx
+	require.NoError(t, err)
+	_, err = e.Chains[tenv.HomeChainSel].Confirm(tx)
+	require.NoError(t, err)
+
+	// Mint a destination token
+	tx, err = dstToken.Mint(
+		e.Chains[tenv.FeedChainSel].DeployerKey,
+		e.Chains[tenv.FeedChainSel].DeployerKey.From,
+		new(big.Int).Mul(twoCoins, big.NewInt(10)),
+	)
+
+	// Confirm the mint tx
+	require.NoError(t, err)
+	_, err = e.Chains[tenv.FeedChainSel].Confirm(tx)
+	require.NoError(t, err)
+
+	linkToken := state.Chains[tenv.HomeChainSel].LinkToken
+
+	// Grant mint and burn roles to the deployer key for the newly deployed linkToken
+	// Since those roles are not granted automatically
+	tx, err = linkToken.GrantMintAndBurnRoles(e.Chains[tenv.HomeChainSel].DeployerKey, e.Chains[tenv.HomeChainSel].DeployerKey.From)
+	require.NoError(t, err)
+	_, err = e.Chains[tenv.HomeChainSel].Confirm(tx)
+	require.NoError(t, err)
+
+	// Mint link token and confirm the tx
+	tx, err = linkToken.Mint(
+		e.Chains[tenv.HomeChainSel].DeployerKey,
+		e.Chains[tenv.HomeChainSel].DeployerKey.From,
+		new(big.Int).Mul(twoCoins, big.NewInt(10)),
+	)
+	require.NoError(t, err)
+
+	_, err = e.Chains[tenv.HomeChainSel].Confirm(tx)
+	require.NoError(t, err)
+
+	t.Run("Send Token Pay with Link token home chain -> remote", func(t *testing.T) {
```

Comment:
q: are these test cases runnable separately or do they depend on each other?

Reply:
They can be run independently